posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -262,9 +263,18 @@ async def handle_model_ready(model: ModelNode, team_id: int, queue: asyncio.Queu
     try:
         if model.selected is True:
             team = await database_sync_to_async(Team.objects.get)(id=team_id)
-            await materialize_model(model.label, team)
+            # Get the workflow ID from the activity info
+            workflow_id = temporalio.activity.info().workflow_id
+            key, delta_table, job_id = await materialize_model(model.label, team, workflow_id)
```

Comment:
I'd move the job model creation out of `materialize_model` - otherwise if the materialization fails, the job will never get updated to `FAILED` cos `job_id` will never get returned